[EPPETA-65] User Creation flow updates

jleiva-gap opened the pull request:
- Update create account, so it does not add a default role to new users, except if the flag to create users as adminstrator is true
- Update Account control to add or remove roles to an user
- New Users are created without role
- Enable navigation menu
- Update Authorization by role
  - Menu options
  - Pages
- Add an user list for administrator
- Add a page to update user roles.
- Add support for multiple roles by user
- Update initial page:
  - Mentor and Supervisor: Evaluation List.
  - Administrator: User list.
  - User without role: Access denied message

jleiva-gap committed:
EPPETA-65 User Creation flow updates

- Update API to edit user roles.
- Add authorization by role.
- Enable Navigation Menu
- Add page to show user list
- Add page to show user profile and edit roles.

jleiva-gap committed:
Update authorization by role

Use role enum
Create user without role.
Use main page for Adminstrator and other users

jleiva-gap committed:
Update SortableTable.js

Remove Input to fix linting error

stevenarnold-ed-fi commented on src/reactapp/src/components/Navbar.js:
```diff
@@ -290,44 +291,51 @@ const MobileNavItem = ({ label, children, href }) => { // Remove type annotation
+    ],
+  },
+  {
+    label: 'User',
```
'Administration' instead of 'User'

stevenarnold-ed-fi commented on src/reactapp/src/components/Navbar.js:
```diff
@@ -290,44 +291,51 @@ const MobileNavItem = ({ label, children, href }) => { // Remove type annotation
-    //    href: '#',
-    //},
+  {
+    label: 'Evaluation',
```
Can we hide or not have this section. With adding a way to get back to the main page in ETA-79, I don't think this as a menu item is desired.

jleiva-gap replied:
I added a redirect to the home page using the image, an I am working on the complete solution in EPPETA-79, but we have a way to get back.

stevenarnold-ed-fi commented on src/reactapp/src/components/AuthenticatedRoute.js:
```diff
@@ -32,10 +32,17 @@ const AuthenticatedRoute = ({ element: authenticatedComponent, ...rest }) => {
-    }
-    );
+      if (roles && !roles.includes('*') && !isLoggedInUserInRole(roles)) {
+        setAlertMessageText("You do not have access to the requested page");
```
Please replace with "You do not have access to the requested page, contact your administrator if you think this is in error."

stevenarnold-ed-fi commented on src/reactapp/src/components/Navbar.js:
```diff
@@ -290,44 +291,51 @@ const MobileNavItem = ({ label, children, href }) => { // Remove type annotation
     children?: Array<NavItem>;
     href?: string;
 }*/
+const showMenuByRol = () => {
```
rename to showMenuByRole

stevenarnold-ed-fi commented:
@jleiva-gap Forgot to add: This doesn't seem to cause any usability issues, but the User Administration page is throwing errors
![2024-01-19_14-15-24](https://github.com/Ed-Fi-Exchange-OSS/EvaluationTracker/assets/48261255/963c9392-dea5-47b4-b3d9-3a5568d48d28)

jleiva-gap committed:
Update based on Comments

Remove menu Evaluations
Update image to redirect to main
Add id to List from sortableTable.
Fix function name typo
Rename User menu to Administration
